pkg/virtualmachine/enricher/singleton.go

```diff
@@ -0,0 +1,42 @@
+package enricher
+
+import (
+	"context"
+
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/pkg/scanners/scannerv4"
+	"github.com/stackrox/rox/pkg/scannerv4/client"
+	"github.com/stackrox/rox/pkg/sync"
+)
+
+var (
+	vmEnricherOnce     sync.Once
+	vmEnricherInstance VirtualMachineEnricher
+	vmLog              = logging.LoggerForModule()
+)
+
+func Singleton() VirtualMachineEnricher {
+	vmEnricherOnce.Do(func() {
+		scannerClient, err := createScannerV4Client()
+		if err != nil {
+			vmLog.Errorf("Failed to create Scanner V4 client for VM enricher: %v", err)
+			// Return enricher with nil client - it will handle errors gracefully
+			vmEnricherInstance = New(nil)
+			return
+		}
+		vmEnricherInstance = New(scannerClient)
+	})
+	return vmEnricherInstance
+}
+
+func createScannerV4Client() (client.Scanner, error) {
+	// Use same defaults as Scanner V4 node scanner but namespace-independent
+	indexerEndpoint := scannerv4.DefaultIndexerEndpoint
+	matcherEndpoint := scannerv4.DefaultMatcherEndpoint
```

Comment:
Propagating comment from #16609 
> I think these are, in theory, configurable. Within Scanner V4 itself, we read a ConfigMap to check the URL of the other components. In other places, we use environment variables. I think it may be safer to rely on env vars instead of this hardcoding.
> 
> For example, I have heard of a user whose org does not allow for the `.svc` suffix

Reply:
exactly! I see in other places we wrap the client into `scannerv4` struct. I think this is good for now as long as it works, but the other scanner clients accept options that won't be applied here, so it may happen that we miss some setting that may apply to some exotic config.

Reply:
I'm not sure how much work it is to implement the use of scannerv4 integration to configure the virtual machine enricher. I suggest this as a follow-up with https://issues.redhat.com/browse/ROX-31177